src/cudamatrix/cu-device.h

```diff
@@ -55,7 +55,12 @@ class CuDevice {
   inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
 
   inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
-    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
+    if (random_stride_mode_) {
+      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
```

Comment:
I actually question the usage of 128, as CUDA allocations tend to be aligned on some multiple of 2 (like 64) anyway. I would exepct that 129 would be more likely to expose buggy code.